### ISSUE TITLE
plainToClass fails for missing complex properties in target class

### ISSUE THREAD

**Issue by apolci** [typestack/class-transformer#88]
Hi,

I have this problem both with plainToClass and deserialize (we use class-transformer do deserialize json).
If the json include some simple property (a string for example) that is not mapped in the class, everything is ok, but if there is a missing complex property (an object) we get an error.
Consider this example:

```
    class TestMissingProperty {
      prop1: string;
    }

    const json = {
      "prop1": "value1",
      "prop2": {
        "anotherProp": "value2"
      }
    };

    const result = plainToClass(TestMissingProperty, json);
```

We get this (quite obscure) error:

```
	TypeError: Cannot read property 'prototype' of undefined
	    at http://localhost:9876/_karma_webpack_/webpack://my-project/~/class-transformer/metadata/MetadataStorage.js:131:70
	    at Array.filter (<anonymous>)
	    at MetadataStorage.webpackJsonp.../../../../class-transformer/metadata/MetadataStorage.js.MetadataStorage.findMetadatas /my-project/~/class-transformer/metadata/MetadataStorage.js:131:1)
	    at MetadataStorage.webpackJsonp.../../../../class-transformer/metadata/MetadataStorage.js.MetadataStorage.findTransformMetadatas /my-project/~/class-transformer/metadata/MetadataStorage.js:35:1)
	    at TransformOperationExecutor.webpackJsonp.../../../../class-transformer/TransformOperationExecutor.js.TransformOperationExecutor.applyCustomTransformations /my-project/~/class-transformer/TransformOperationExecutor.js:182:1)
	    at _loop_1 /my-project/~/class-transformer/TransformOperationExecutor.js:149:1)
	    at TransformOperationExecutor.webpackJsonp.../../../../class-transformer/TransformOperationExecutor.js.TransformOperationExecutor.transform /my-project/~/class-transformer/TransformOperationExecutor.js:172:1)
	    at _loop_1 /my-project/~/class-transformer/TransformOperationExecutor.js:148:1)
	    at TransformOperationExecutor.webpackJsonp.../../../../class-transformer/TransformOperationExecutor.js.TransformOperationExecutor.transform /my-project/~/class-transformer/TransformOperationExecutor.js:172:1)
	    at _loop_1 /my-project/~/class-transformer/TransformOperationExecutor.js:148:1)
```

Is there a way to make that class-transformer don't fail in this case?
The only way I've found is to explicitly map or @Exclude the additional properties, but would like to be more flexible and avoid to map non interesting properties we get in the json.

Is there a way to get a more meaningful error, that allow to understand which is the offending property? When we have complex objects with many properties it's quite time consuming to look for the one that has been forgotten.

Thank you

**Comment by Farcek** [comment]
me to

**Comment by lavavrik** [comment]
Looks like still an issue...

@apolci Did you managed to solve this?

**Comment by NoNameProvided** [comment]
Hi guys! 

Can you give me some example code what fails? I just have added a test for @apolci case in ac92126 and it transforms the object successfully as you can see in our [Travis](https://travis-ci.org/typestack/class-transformer/builds/337182425)

**Comment by NoNameProvided** [comment]
Just published `0.1.9`, please test it and report if you still encounter this issue. Thanks!

**Comment by NoNameProvided** [comment]
Closing this as solved.
